### ISSUE TITLE
Error message for indentation mistake ambiguous

### ISSUE THREAD

**Issue by julianbarg** [mikefarah/yq#1186]
**Describe the bug**
Wrong indentation leads to the following error: 
```Error: bad file '/home/data1.yaml': yaml: line 2: could not find expected ':'```

Version of yq: 4.24.2
Operating system: linux/ubuntu 20.04
Installed via: snap

**Input Yaml**
data1.yml:
```yaml
key: |
mistake
```

**Command**
The command you ran:
```
yq ".key" $yaml
```

**Actual behavior**
Results in the error:
```Error: bad file '/home/data1.yaml': yaml: line 2: could not find expected ':'```

**Expected behavior**
Something along the lines of "Did you mean to indent line 2". Or "Line 2 indented incorrectly".

**Additional context**
From everything I have seen online, indentation errors in YAML are quite common, so making them easier to debug should be a no-brainer.

**Comment by mikefarah** [comment]
I agree - however, the error message is straight from the underlying yaml parser (go-yaml) - feel free to raise an issue there

**Comment by mikefarah** [comment]
Actually I found an existing issue for it: https://github.com/go-yaml/yaml/issues/727
